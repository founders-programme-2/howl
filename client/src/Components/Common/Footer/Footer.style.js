import styled from 'styled-components';
import '../../../globalStyle';

import responsiveSizes from '../../../constants/responsiveSizes';

const { MDM } = responsiveSizes;

export const FooterDiv = styled.footer`
  min-width: 100%;
  background-color: var(--secondary-bg-color);
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  bottom: 0;
  position: relative;
  align-items: center;
  @media (max-width: ${MDM}) {
    flex-flow: column nowrap;
  }
`;

export const TitleDiv = styled.section`
  width: 30%;
  height: 100%;
  @media (max-width: ${MDM}) {
    width: 100%;
    height: 17vh;
  }
`;

export const StyledTitle = styled.h2`
  color: var(--secondary-font-color);
  text-align: center;
  text-transform: uppercase;
  margin: 5% 0 0 0;
`;

export const FilterDiv = styled.section`
  width: 50%;
  height: 100%;
  display: flex;
  flex-direction: row;
  margin: auto;
  @media (max-width: ${MDM}) {
    width: 100%;
    margin: 4% auto;
    display: flex;
    flex-flow: column nowrap;
  }
`;

export const SearchDiv = styled.section`
  width: 25%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin: auto;

  @media (max-width: ${MDM}) {
    width: 100%;
    margin-bottom: 4%;
  }
`;

export const ButtonsContainer = styled.section`
  width: 82%;
  margin: 1% auto;
  display: flex;
  justify-content: center;
  @media (max-width: ${MDM}) {
    width: 96.6%;
    margin: 0 auto 1.7%;
  }
`;

export const BoldText = styled.span`
  font-weight: bold;
`;
