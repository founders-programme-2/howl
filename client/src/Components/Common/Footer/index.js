import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import {
  Button,
  Select,
  MenuItem,
  FormControl,
  TextField,
  withStyles,
  Input,
  InputLabel,
  Chip,
  Checkbox,
  ListItemText,
  MuiThemeProvider,
} from '../../muIndex';
import {
  FooterDiv,
  TitleDiv,
  StyledTitle,
  FilterDiv,
  SearchDiv,
  BoldText,
  ButtonsContainer,
} from './Footer.style';
import { tags as tagsArr, categories, locations } from '../../data';
import { styles, BootstrapInput, MenuProps, footerTheme } from './mui.style';
import navigationUrls from '../../../constants/navigationUrls';

const renderSelectItems = items => {
  return items.sort().map(item => (
    <MenuItem key={item} value={item}>
      {item}
    </MenuItem>
  ));
};

const renderMultiSelectItems = (items, selectedItems) => {
  return items.sort().map(item => (
    <MenuItem key={item} value={item}>
      <Checkbox checked={selectedItems.indexOf(item) > -1} />
      <ListItemText primary={item} />
    </MenuItem>
  ));
};

const yearsGenerator = () => {
  const years = [];
  for (let i = 1960; i <= 1990; i += 1) {
    years.push(i);
  }
  return years;
};

class Footer extends Component {
  state = {
    tags: [],
    category: null,
    year: null,
    location: null,
    search: null,
    result: [],
  };

  submitSearch = ({ tags, category, year, location, search }) => {
    const { setFilters, history } = this.props;
    const { RESULTS_URL } = navigationUrls;
    setFilters({ tags, category, year, location, search });
    history.push(RESULTS_URL);
  };

  updateStateValue = (name, value) => {
    this.setState({ [name]: value });
  };

  howToSearchHandler = () => {
    const { history } = this.props;
    if (history) history.push('/help');
  };

  resetFields = () => {
    this.setState(
      {
        tags: [],
        category: null,
        year: null,
        location: null,
        search: null,
      },
      () => {
        const { history } = this.props;
        history.push('/archive');
      }
    );
  };

  render() {
    const { classes } = this.props;
    const { tags, category, year, location, search } = this.state;
    return (
      <MuiThemeProvider theme={footerTheme}>
        <FooterDiv>
          <TitleDiv>
            <StyledTitle>Search Archive</StyledTitle>
            <Button
              className={classes.howToSearchButton}
              onClick={this.howToSearchHandler}
            >
              How to search
            </Button>
          </TitleDiv>

          <FilterDiv>
            <FormControl className={classes.formControl}>
              <Select
                className={classes.select}
                value={category || 'Category'}
                aria-label="category"
                onChange={event => {
                  this.updateStateValue('category', event.target.value);
                }}
                input={
                  <BootstrapInput
                    name="category"
                    id="category-customized-select"
                    aria-label="Category"
                  />
                }
              >
                <MenuItem value="Category">
                  <BoldText>Category</BoldText>
                </MenuItem>
                {renderSelectItems(categories)}
              </Select>

              <Select
                className={classes.select}
                value={location || 'Location'}
                aria-label="location"
                onChange={event => {
                  this.updateStateValue('location', event.target.value);
                }}
                input={
                  <BootstrapInput
                    name="location"
                    id="location-customized-select"
                    aria-label="Location"
                  />
                }
              >
                <MenuItem value="Location">
                  <BoldText>Location</BoldText>
                </MenuItem>
                {renderSelectItems(locations)}
              </Select>
              <Select
                className={classes.select}
                value={year || 'Year'}
                aria-label="year"
                onChange={event => {
                  this.updateStateValue('year', event.target.value);
                }}
                input={
                  <BootstrapInput
                    name="year"
                    id="year-customized-select"
                    aria-label="Year"
                  />
                }
              >
                <MenuItem value="Year">
                  <BoldText>Year</BoldText>
                </MenuItem>
                {renderSelectItems(yearsGenerator())}
              </Select>
            </FormControl>
            <FormControl
              variant="filled"
              className={classes.multipleSelectFormControl}
            >
              <InputLabel
                htmlFor="selectMultipleCheckbox"
                className={classes.MultipleSelectInputLabel}
              >
                Tags
              </InputLabel>
              <Select
                multiple
                disableUnderline
                value={tags}
                aria-label="tags"
                onChange={event => {
                  this.updateStateValue('tags', event.target.value);
                }}
                input={<Input id="selectMultipleCheckbox" aria-label="Tags" />}
                renderValue={selected => (
                  <div className={classes.chips}>
                    {selected.map(value => (
                      <Chip
                        key={value}
                        label={value}
                        className={classes.chip}
                      />
                    ))}
                  </div>
                )}
                MenuProps={MenuProps}
              >
                {renderMultiSelectItems(tagsArr, tags)}
              </Select>
            </FormControl>
          </FilterDiv>

          <SearchDiv>
            <FormControl className={classes.searchFormControl}>
              <TextField
                id="outlined-search"
                label="Search..."
                value={search || ''}
                onChange={event => {
                  this.updateStateValue('search', event.target.value);
                }}
                InputLabelProps={{
                  className: classes.searchTextFieldLabel,
                }}
                type="search"
                className={classes.searchTextField}
                InputProps={{
                  disableUnderline: true,
                }}
              />
            </FormControl>
            <ButtonsContainer>
              <Button
                className={classes.searchButton}
                onClick={() => {
                  this.submitSearch(this.state);
                }}
              >
                Search
              </Button>
              <Button
                className={classes.resetButton}
                onClick={this.resetFields}
              >
                Reset
              </Button>
            </ButtonsContainer>
          </SearchDiv>
        </FooterDiv>
      </MuiThemeProvider>
    );
  }
}

export default withRouter(withStyles(styles)(Footer));
