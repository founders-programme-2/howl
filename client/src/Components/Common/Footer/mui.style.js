import { InputBase, withStyles, createMuiTheme } from '../../muIndex';

export const footerTheme = createMuiTheme({
  palette: {
    primary: {
      main: '#F6C25B',
    },
    secondary: {
      main: '#F6C25B',
    },
  },
});

export const styles = theme => ({
  howToSearchButton: {
    textTransform: 'uppercase',
    backgroundColor: 'var(--main-btn-color)',
    width: '50%',
    margin: `5% 25% 5% 25%`,
    fontFamily: ['Raleway', 'sans-serif'].join(','),
    fontWeight: 'bold',
    borderRadius: '20px',
    '&:hover': {
      color: theme.palette.common.white,
      backgroundColor: 'var(--main-btn-color)',
    },
    '@media (max-width:780px)': {
      margin: `1% 25% 1% 25%`,
    },
  },
  formControl: {
    width: '100%',
    flexDirection: 'row',
    flexWrap: 'nowrap',
    justifyContent: 'space-evenly',
  },

  multipleSelectFormControl: {
    width: '95.5%',
    margin: '2.5%',
    borderRadius: '4px',
    backgroundColor: 'white',
    padding: '4px',
  },
  chips: {
    width: '100%',
    marginTop: '2.5%',
    marginBottom: '2.5%',
  },
  chip: {
    width: '100%',
    marginTop: '2.5%',
    marginBottom: '5%',
  },
  MultipleSelectInputLabel: {
    width: '95%',
    margin: '0px',
    color: 'black',
    fontWeight: '600',
    fontFamily: 'Raleway, sans-serif',
    fontSize: 16,
  },
  select: {
    width: '30%',
    marginTop: '2.5%',
  },
  searchFormControl: {
    width: '100%',
    height: '50%',
    margin: '0px auto ',
    display: 'flex',
    justifyContent: 'center',
  },
  searchTextField: {
    margin: 'auto',
    backgroundColor: theme.palette.common.white,
    borderRadius: '4px',
    fontWeight: 'bold',
    width: '80%',
    height: '20%',
  },
  searchTextFieldLabel: {
    fontWeight: 'bold',
  },

  searchButton: {
    width: '60%',
    backgroundColor: 'var(--main-btn-color)',
    borderRadius: '20px',
    fontWeight: 'bold',
    marginTop: '2%',
    marginRight: '2.5%',
    '&:hover': {
      color: theme.palette.common.white,
      backgroundColor: 'var(--main-btn-color)',
    },
  },
  resetButton: {
    width: '30%',
    backgroundColor: 'var(--highlight-color)',
    color: theme.palette.common.black,
    borderRadius: '20px',
    fontWeight: 'bold',
    marginTop: '2%',
    marginLeft: '2.5%',
    '&:hover': {
      color: theme.palette.common.white,
      backgroundColor: 'var(--highlight-color)',
    },
  },
});

export const BootstrapInput = withStyles(theme => ({
  root: {
    'label + &': {
      marginTop: theme.spacing(3),
    },
  },
  input: {
    borderRadius: 4,
    position: 'relative',
    backgroundColor: theme.palette.background.paper,
    border: '1px solid #ced4da',
    fontSize: 16,
    width: '80%',
    padding: '5% 16% 5% 8%',
    transition: theme.transitions.create(['border-color', 'box-shadow']),
    fontFamily: 'Raleway, sans-serif',
    fontWeight: '600',
    '&:focus': {
      borderRadius: 4,
      borderColor: '#80bdff',
      boxShadow: theme.palette.background.paper,
      backgroundColor: theme.palette.background.paper,
    },
    svg: {
      width: '20%',
    },
  },
}))(InputBase);

const ITEM_HEIGHT = 48;
const ITEM_PADDING_TOP = 8;
export const MenuProps = {
  PaperProps: {
    style: {
      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
      width: 250,
    },
  },
};
