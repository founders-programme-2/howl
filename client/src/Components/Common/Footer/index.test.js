import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import Footer from './index';

Enzyme.configure({ adapter: new Adapter() });

const { mount } = Enzyme;

describe('Footer Component Test', () => {
  // Render Footer component for testing it, Nest it in <BrowserRouter> because we can't use <Link> outside the router! (Link is in Footer).
  const footer = mount(
    <BrowserRouter>
      <Footer />
    </BrowserRouter>
  );

  // Normal test case, it's just to check if the rendered component output is the same as the one on the snapshot or not!
  test('Natural', () => {
    expect(footer).toMatchSnapshot();
  });

  // To test category select box if it's working correctly.
  test('Category Select check', () => {
    // Find the select category, that contains the select value in it's text.
    const select = footer.find({ id: 'select-category' });

    // Simulate the click event on the select category box.
    select.simulate('click');

    // Find choice of categorym and simulate it's click (to choose new select value).
    footer
      .find({ role: 'option', 'data-value': 'Campaigns' })
      .at(0)
      .simulate('click');

    // Check the element that contains the value in its html text, if its text change to be equal a value or not.
    expect(select.text()).toBe('Campaigns');
  });

  // And so on for the other tests...
  test('Location Select check', () => {
    const select = footer.find({ id: 'select-location' });
    select.simulate('click');
    footer
      .find({ role: 'option', 'data-value': 'London' })
      .at(0)
      .simulate('click');
    expect(select.text()).toBe('London');
  });

  test('Year Select check', () => {
    const select = footer.find({ id: 'select-year' });
    select.simulate('click');
    footer
      .find({ role: 'option', 'data-value': 1970 })
      .at(0)
      .simulate('click');
    expect(select.text()).toBe('1970');
  });
});
