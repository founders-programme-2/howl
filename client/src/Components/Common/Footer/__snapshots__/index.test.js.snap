// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Test Natural 1`] = `ReactWrapper {}`;
